File: src/main/java/gregtechlite/gtlitecore/client/renderer/TranscendentRenderBehavior.java
package gregtechlite.gtlitecore.client.renderer;

import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;

public interface TranscendentRenderBehavior extends ItemRendererManager
{

    float getRotationSpeed();

    float getRotationAxisX();

    float getRotationAxisY();

    float getRotationAxisZ();

    float getFloatingOffset();

    @SideOnly(Side.CLIENT)
    default float[] getRotationAxis()
    {
        return new float[] { getRotationAxisX(), getRotationAxisY(), getRotationAxisZ() };
    }

}

File: src/main/kotlin/gregtechlite/gtlitecore/client/renderer/TranscendentRenderBehavior.kt
package gregtechlite.gtlitecore.client.renderer

import net.minecraftforge.fml.relauncher.Side
import net.minecraftforge.fml.relauncher.SideOnly

interface TranscendentRenderBehavior : ItemRendererManager
{
    @SideOnly(Side.CLIENT)
    fun getRotationSpeed(): Float

    @SideOnly(Side.CLIENT)
    fun getRotationAxisX(): Float

    @SideOnly(Side.CLIENT)
    fun getRotationAxisY(): Float

    @SideOnly(Side.CLIENT)
    fun getRotationAxisZ(): Float

    @SideOnly(Side.CLIENT)
    fun getFloatingOffset(): Float

    @SideOnly(Side.CLIENT)
    fun getRotationAxis(): FloatArray = floatArrayOf(getRotationAxisX(), getRotationAxisY(), getRotationAxisZ())
}